DevelopmentProposals/readme.md

```diff
@@ -0,0 +1,12 @@
+## Overview
+
+This directory contains all the design documents for OpenSim.
```

Comment:
For consistency with the directory name, replace *design document* with *development proposal* throughout this PR.  (I think it would also be good to have design documents at some point in the future as separate from the dev props.)